src/components/dataItem/CoordinateField.js

```diff
@@ -1,98 +1,59 @@
-import { connect } from 'react-redux'
-import {
-    loadProgramTrackedEntityAttributes,
-    loadProgramStageDataElements,
-} from '../../actions/programs.js'
+import React, { useEffect } from 'react'
+import { useEventDataItems } from '../../hooks/useEventDataItems.js'
 import { SelectField } from '../core/index.js'
 
-class CoordinateField extends Component {
-    static propTypes = {
-        dataElements: PropTypes.object.isRequired,
-        loadProgramStageDataElements: PropTypes.func.isRequired,
-        loadProgramTrackedEntityAttributes: PropTypes.func.isRequired,
-        programAttributes: PropTypes.object.isRequired,
-        onChange: PropTypes.func.isRequired,
-        className: PropTypes.string,
-        program: PropTypes.object,
-        programStage: PropTypes.object,
-        value: PropTypes.string,
-    }
-
-    componentDidMount() {
-        this.loadData()
-    }
-
-    componentDidUpdate(prevProps) {
-        const { program, onChange } = this.props
-
-        if (program !== prevProps.program) {
-            onChange('event')
-        }
-
-        this.loadData()
-    }
-
-    render() {
-        const {
-            value,
-            program,
-            programStage,
-            programAttributes,
-            dataElements,
-            onChange,
-            className,
-        } = this.props
-        let fields = [
-            { id: 'event', name: i18n.t('Event location') }, // Default coordinate field
-        ]
-
-        if (program && programStage) {
-            fields = fields.concat(
-                [
-                    ...(programAttributes[program.id] || []),
-                    ...(dataElements[programStage.id] || []),
-                ].filter((field) => field.valueType === 'COORDINATE')
-            )
-        }
-
-        return (
-            <SelectField
-                label={i18n.t('Coordinate field')}
-                items={fields}
-                value={fields.find((f) => f.id === value) ? value : 'event'}
-                onChange={(field) => onChange(field.id)}
-                className={className}
-            />
-        )
-    }
-
-    loadData() {
-        const {
-            program,
-            programStage,
-            programAttributes,
-            dataElements,
-            loadProgramTrackedEntityAttributes,
-            loadProgramStageDataElements,
-        } = this.props
-
-        if (program && !programAttributes[program.id]) {
-            loadProgramTrackedEntityAttributes(program.id)
+const EVENT_COORDINATE_FIELD_ID = 'event'
+
+const includeTypes = ['COORDINATE']
+
+const CoordinateField = ({
+    value,
+    program,
+    programStage,
+    onChange,
+    className,
+}) => {
+    const { eventDataItems } = useEventDataItems({
+        programId: program?.id,
+        programStageId: programStage?.id,
+        includeTypes,
+    })
+
+    useEffect(() => {
+        if (program) {
+            onChange(EVENT_COORDINATE_FIELD_ID)
```

Comment:
There is a bug here: If I change the coordinate field to "Household location" for "Inpatient morbidity and mortality" and add the layer (the layer is currently not showing due to a backend bug: https://dhis2.atlassian.net/browse/DHIS2-13528?focusedCommentId=179910), then edit the layer again - the coordinate field is not keeping its value.